src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy

```diff
@@ -35,44 +35,45 @@ public class QualityAwareJavaPlugin implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
-        project.plugins.apply(JAVA_PLUGIN_ID)
-
         project.extensions.create(QualityAwareJavaExtension.NAME, QualityAwareJavaExtension)
+        project.childProjects.values().each { subProject ->
```

Comment:
why don't use `subprojects` for quality plugin? Something like that

subprojects {
apply plugin: 'java'
...
}